src/libcadet/model/parts/ConvectionDispersionOperator.cpp

```diff
@@ -185,29 +186,25 @@ bool ConvectionDispersionOperatorBase::configure(UnitOpIdx unitOpIdx, IParameter
  */
 bool ConvectionDispersionOperatorBase::notifyDiscontinuousSectionTransition(double t, unsigned int secIdx)
 {
-	double prevVelocity = static_cast<double>(_curVelocity);
-
 	// If we don't have cross section area, velocity is given by parameter
+	double dir_old = _dir;
 	if (_crossSection <= 0.0)
-		_curVelocity = getSectionDependentScalar(_velocity, secIdx);
+	{
+		double dir_new = static_cast<double>(getSectionDependentScalar(_velocity, secIdx));
+		_dir = (dir_new >= 0);
```

Comment:
This will result in `_dir` having the value either `0` (test fails) or `1` (test succeeds). We need `-1` or `1`, so I'd suggest
```cpp
_dir = (dir_new >= 0.0) ? 1 : -1;
```

Reply:
I see my mistake. Can you please elaborate on the syntax of your suggestion?

Reply:
This `?` is called a *ternary operator*. It's similar to Python's
```python
a = 2 if b == 3 else 6
```
which would translate to
```cpp
int a = (b == 3) ? 2 : 6;
```
So it's `(condition) ? trueValue : falseValue;`.

Reply:
That's what I expected but I really was not sure about the operator. Thanks for explaining!